src/main/java/com/klodnicki/Bike/v2/service/interfacee/GenericRentalActionService.java

```diff
@@ -0,0 +1,9 @@
+package com.klodnicki.Bike.v2.service.interfacee;
+
+public interface GenericRentalActionService<G, H, T, ID, ID2, ID3> {
```

Comment:
My suggestion:
```suggestion
public interface RentService<G, H, T, ID, ID2, ID3> {
```
That way it has cleaner name and it's consistent with other interfaces of same type.

As for all those generic types that are introduced (6 of them!) it looks... Kinda sketchy. It's easy to lose the track of it. From my quick eyeballing this issue I think at least the returnVehicle method could take a rework in number of parameters it takes.
I think it'd be possible to only have one parameter which wouldn't even need to be generic.

Reply:
1. days of rent is the information provided by user for how long he wants to rent a vehicle
2. name RentService sound good to me! I thought that because it holds generic methods it should have a `generic` name inside. But okay! for me RentService sounds good!
3. About returnVehicle(), you think that one parameter would be a RentRequest?

Reply:
Regarding ad. 1: User can rent a vehicle without specifying for how long he'll have a bike, because application should allow to rent a vehicle with price per minute rented **or** to rent it per days with static price
ad. 2: I believe what this method needs is just a user which asks to return a vehicle - in this case, a bike. As for the structure of the app, you can't have more than one vehicle rented. So you just click return, the controller gets the user which sends the request to return a bike (obviously that user has to be logged in) and that what you're passing to that method. Which bike to return you know, because User class has that specified.

Reply:
Ad.1 I understand that application counts the time and rental cost. But please tell me why then there is a task to update time period of renting a vehicle? 

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/63dc96e3-eaa6-4016-9035-71b66a5b3601)

Reply:
Ad 3. What about station? If there is one parameter (lets say user id) for returnVehicle() how application will know to which station bike was returned?

Reply:
> Ad.1 I understand that application counts the time and rental cost. But please tell me why then there is a task to update time period of renting a vehicle?

Okay, so I'll explain the whole user story here:
User clicks to rent a bike and have two options. 1st to just rent it and it's going to be charged per minutes spent riding the bike (or just having it rented, without even riding it, but you get the idea). 2nd option is to pick option i.e. pay 20 PLN, get a bike (or other vehicle) for 24 hours. Before the time runs out you have to return the bike to any station.
Now we get to the updating the rent time. User can decide that they want to rent it for longer than initially they thought. So they have a rent still going for 24 hours, but they want to rent it for the whole weekend. So another 24 hours is needed. What they do is just clicks to change the rent from 24h to 48h and that's what this endpoint is all about.

> Ad 3. What about station? If there is one parameter (lets say user id) for returnVehicle() how application will know to which station bike was returned?

You're absolutely correct! I've missed it. But the returnVehicle method can still have one parameter. It just needs to be some kind of DTO (as you said earlier) which will hold the user that is requesting the change and a station id where the bike is being returned. This will simplify those multiple generic types!

That's because you could even make it not generic for this specific case. But I'm not yet fully convinced about that idea, I'd need to see it implemented or just put more thought time into it :)

Reply:
Ad.1 I see this concept! So I hope that you have answered yourself why field `daysOfRent` is in Rent entity then!

Reply:
Ad.1 this is my proposal of counting rental cost based on the concept which you presented:
![Uploading image.png…]()

Reply:
Ad.3 Ok I made un upgrade of returnVehicle(). Now you only need 2 parameters. Rent id and station id to where user returns a bike. One is @PathVariable, the second is @RequestParam therefore all necessary information is provided into URL so I thinks its very convenient way.
localhost:8080/api/returns/1?returnChargingStationId=1